### PR TITLE
Feature/init

### LINKED ISSUE #1: Model
- id bigint
- uuid varchar32 
- group  varchar100
- key  varchar100
- value mediumtext
- public boolean
- enumerable (show in settings endpoint)
- type varchat12 (markdown, json, image, etc), config value taken from laravels `config()` 
- sort int

### PR DESCRIPTION
**Pull request by qunabu**
Closes #1

### CODE REVIEW

File: src/Http/Requests/SettingsCreateRequest.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function rules()
+    {
+        return [
+            'key' => ['required', 'string'],
+            'group' => ['required', 'string'],
+            'public' => ['boolean'],
+            'enumerable' => ['boolean'],
+            'sort' => ['integer'],
+            'type' => ['required', 'in:' . implode(',', SettingsTypes::getValues())],
```
Comment (pa-cholek) [inline src/Http/Requests/SettingsCreateRequest.php]:
```php
use Illuminate\Validation\Rule;

['required', Rule::in(SettingsTypes::getValues())]
```

File: src/Http/Resources/SettingsCollection.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        foreach ($this->collection as $value) {
+            switch($value->type) {
+                case "config":
+                    $data[$value->group][$value->key] = config($value->value);
+                    break;
+                case "json":
+                    $data[$value->group][$value->key] = json_decode($value->value);
+                    break;
+                default:
+                $data[$value->group][$value->key] = $value->value;
```
Comment (pa-cholek) [inline src/Http/Resources/SettingsCollection.php]: Wouldn't it be possible to use custom Caster in Model for casting this value to correct type?

https://laravel.com/docs/8.x/eloquent-mutators#custom-casts

File: src/Http/Resources/SettingsCollection.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     * @return array
+     */
+    public function toArray($request)
+    {
+        $data = [];
+
+        foreach ($this->collection as $value) {
+            switch($value->type) {
+                case "config":
+                    $data[$value->group][$value->key] = config($value->value);
```
Comment (pa-cholek) [inline src/Http/Resources/SettingsCollection.php]: I'm not really sure what's the intended use case of "config" type setting?

Comment (pa-cholek) [reply src/Http/Resources/SettingsCollection.php]: If "config" type settings will be used to overwrite Laravel Config values, then this probably should retrieve `config($value->key)` and not value?

**Review by pa-cholek — CHANGES_REQUESTED** [review]

File: src/Http/Resources/SettingsCollection.php (hunk 3 above)
Comment (qunabu) [reply src/Http/Resources/SettingsCollection.php]: true, but `value` cannot be empty. example stripe publish key, smtp email to, etc

**Review by pa-cholek — APPROVED** [review]

**Pull request merged**